react/ConfirmationTitle.tsx

```diff
@@ -2,16 +2,34 @@ import React, { FC } from 'react'
 import { FormattedMessage } from 'react-intl'
 import { useCssHandles } from 'vtex.css-handles'
 
+import { useOrderGroup } from './components/OrderGroupContext'
+
 const CSS_HANDLES = ['confirmationTitle']
 
-const ConfirmationTitle: FC = () => {
+interface Props {
+  displayName?: boolean
+}
+
+const ConfirmationTitle: FC<Props> = ({ displayName = false }) => {
   const handles = useCssHandles(CSS_HANDLES)
+  const orderGroup = useOrderGroup()
+  const profile = orderGroup.orders[0].clientProfileData
 
   return (
     <h4
       className={`${handles.confirmationTitle} tc c-on-base mt7 mb0 t-heading-4`}
     >
-      <FormattedMessage id="store/header.thanks" />
+      {displayName ? (
+        <FormattedMessage
+          id="store/header.thanks.name"
+          values={{
+            firstName: profile.firstName,
+            lastName: profile.lastName,
+          }}
+        />
+      ) : (
+        <FormattedMessage id="store/header.thanks" />
+      )}
```

Comment:
There's no need for the prop `displayName`, we can just provide the values and the user chooses in the translation message if they want to use it the `firstName` and `lastName`.

```suggestion
      <FormattedMessage
        id="store/header.thanks"
        values={{
          firstName: profile.firstName,
          lastName: profile.lastName,
        }}
      />
```

Reply:
Removed the prop and will handle this from the translation itself.

Reply:
Also removed the extra message I added ("store/header.thanks.name")